### PR TITLE
Modify S109: Migrate To LayC

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [x] logical errors and incorrect information
- [x] information gaps and missing content
- [x] text style and tone
- [x] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

**Comment by zsolt-kolbay-sonarsource** [comment]
Is [S109/rule.adoc](https://github.com/SonarSource/rspec/blob/82ea93b92a5bbb431c50b89c82bd0479041fa341/rules/S109/rule.adoc) used anywhere? If not, it should be removed.

File: rules/S109/cfamily/rule.adoc [hunk 1]
```diff
@@ -1,25 +1,18 @@
...
 
-=== Noncompliant code example
+=== Code examples
+
+==== Noncompliant code example
 
-[source,cpp]
+[source,cpp,diff-id=1,diff-type=noncompliant]
 ----
 void doSomething(int var) {
```
[the remark this answers is not in the document]
Comment (rudy-regazzoni-sonarsource) [reply rules/S109/cfamily/rule.adoc:14]: I thought the same at first, but this variable is used: `if (var == 42) {`
I find this example pertinent: in the compliant version, the `var` parameter is not declared as a `int` anymore but as a `Status`, showing a compliant use case with `enum`.
WDYT?

File: rules/S109/cobol/rule.adoc [hunk 2]
```diff
@@ -1,10 +1,45 @@
...
+    DISPLAY "Count:"WS-COUNT.
+    ADD 1 TO WS-COUNT.
+----
+
+==== Compliant solution
+
+[source,cobol,diff-id=1,diff-type=compliant]
+----
+IDENTIFICATION DIVISION.
+PROGRAM-ID. HELLO-WORLD.
```
[the remark this answers is not in the document]
Comment (rudy-regazzoni-sonarsource) [reply rules/S109/cobol/rule.adoc:29]: Good catch! I forgot to remove that part in the compliant example.

File: rules/S109/cfamily/rule.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (zsolt-kolbay-sonarsource) [reply rules/S109/cfamily/rule.adoc:14]: I missed that one. I agree, let's keep it. But to make the Noncompliant code more consistent with the Compliant solution I'd switch the operands of the comparison from
```
if (var == 42)
```
to
```
if (42 == var)
```

File: rules/S109/cfamily/rule.adoc [hunk 3]
```diff
@@ -27,13 +27,13 @@ void doSomething(int var) {
 [source,cpp,diff-id=1,diff-type=compliant]
 ----
 enum Status {
-  STATUS_KO = 0,
+  STATUS_ERROR = 0,
```
Comment (zsolt-kolbay-sonarsource) [inline rules/S109/cfamily/rule.adoc:30]: Nitpick: usually we use 0 for an OK status code and non-zero for errors.

**Review by zsolt-kolbay-sonarsource — APPROVED** [review]
LGTM! Left one nitpicking comment.

**Pull request merged**